ecl/hthor/hthor.cpp

```diff
@@ -5005,7 +5027,7 @@ const void * CHThorSelfJoinActivity::nextInGroup()
         {
             if (!group.isItem(rightIndex))
             {
-                lhs.setown(input->nextInGroup());   //get from dualcache
+                lhs.setown(dualCacheInput->nextInGroup());
                 if (lhs)
                 {
                     rightIndex = 0;
```

Comment:
There is a bug in the code that follows - 

                    const void * ret = joinRecords(lhs, rhs, ++joinCounter, NULL);

could return NULL if the transform contains a skip.  Unlikely and a separate jira.